### PR TITLE
Switching test to use the in memory persistence adapters

### PR DESCRIPTION
**Pull request by stanlemon**
This is an attempt to decouple the `SidelineSpoutTest` from Zookeeper.

### CODE REVIEW

**Comment by stanlemon** [comment]
While digging into this I also tried swapping in the `MockConsumer` and that proved to be much more hairy than I thought it would be. This test is tightly coupled to KafkaRecord instances and it's just going to take some serious time to decouple them and maintain the test.

File: src/main/java/com/salesforce/storm/spout/dynamic/persistence/InMemoryPersistenceAdapter.java [hunk 1]
```diff
@@ -35,27 +35,23 @@
 public class InMemoryPersistenceAdapter implements PersistenceAdapter {
 
     /**
-     * Used within tests to more easily validate assertions.
+     * In memory store for this adapter's data.
      */
-    public static PersistConsumerStateCallback<String, Integer, Long> persistConsumerStateCallback = (consumerId, partitionId, offset) -> {
-        // No-op by default.
-    };
+    public static Map<String, Long> storedConsumerState = Maps.newHashMap();
```
Comment (Crim) [inline src/main/java/com/salesforce/storm/spout/dynamic/persistence/InMemoryPersistenceAdapter.java]: This feels like it'll break as soon as you fire up more than one vspout will it not?  Being static and shared across all instances?

Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/persistence/InMemoryPersistenceAdapter.java]: Probably just need to make this a concurrent map in both adapters to be safe.

**Comment by stanlemon** [comment]
Based upon your comments @Crim I am going to close this for now.

**Pull request closed without merging**